pyop2/base.py

```diff
@@ -2955,15 +3035,31 @@ def offset(self):
         """The vertical offset."""
         return self._offset
 
+    def _constant_layer_masks(self, which):
+        if self.offset is None:
+            return {}
+        idx = {"bottom": -2, "top": -1}[which]
+        masks = {}
+        for method, (section, iset, facet_iset) in six.iteritems(self.boundary_masks):
+            top = facet_iset.indices[idx]
+            off = section.getOffset(top)
+            dof = section.getDof(top)
+            section.getDof(top)
+            indices = iset.indices[off:off+dof]
+            mask = np.zeros(len(self.offset), dtype=IntType)
+            mask[indices] = -1
+            masks[method] = mask
+        return masks
```

Comment:
This is backwards-combat for the constant layers case because I don't want to pick apart that codgen.